core/domain/classifier_domain.py

```diff
@@ -277,6 +277,27 @@ def status(self):
     def training_data(self):
         return self._training_data
 
+    def update_state_name(self, state_name):
```

Comment:
I have same doubt as @seanlip. I don't think we need `state_name` to be mutable in training job model. 

(However we do need it to be mutable in classifier_data_model. We will have to update `state_name` when storing new exp_version does not trigger retraining and we use old classifier data model. So in that case we will have to update `state_name`, I guess.)